src/components/utils/hooks.ts

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+export type UpdateArgument<T extends object> = T | ((previousArg: T) => Partial<T>);
+
+export function useObjectState<T extends object>(
```

Comment:
What is it used for?

Reply:
It is a util to set a state object property conveniently